transport/twirp/client.go

```diff
@@ -0,0 +1,103 @@
+package twirp
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/endpoint"
+)
+
+// rpcFn is a simple type to represent an RPC method on the Twirp client.
+type rpcFn func(context.Context, interface{}) (interface{}, error)
```

Comment:
This type matches the `endpoint.Endpoint` function. Maybe we should just use that? 
It's odd to have the NewClient function require a private type as a param.